db/queries.py

```diff
@@ -1,46 +1,89 @@
+from __future__ import annotations
+
 from db.models import Message, User, Chat
 from django.db.models import Q, Count, F
 
 
 def get_messages_that_contain_word(word: str) -> list[Message]:
-    pass
+    return list(
+        Message.objects.select_related("user", "chat")
```

Comment:
You don’t need `select_related` In this exact case

Reply:
No, I need it because in the __str__ method of the Message model there are references to Chat and User. I've checked in the Django shell plus the queries.

Reply:
You see these queries because you do just this code Message.objects.filter(text__icontains=word) and when it shows on screen it calls str method for each
If you do just:
messages = Message.objects.filter(text__icontains=word)
It will no happens

Reply:
Or please pin a screenshot. Maybe I am wrong here

Reply:
@ArsenPidhoretskyi no, this happens anyway. 
![without select related](https://user-images.githubusercontent.com/114873486/217819743-631661f3-8ad8-4f21-a1d2-c40a7463b481.png)

Reply:
@ArsenPidhoretskyi and here is with select-related - only one query with joins 
![with select related](https://user-images.githubusercontent.com/114873486/217819914-7a715331-f95d-42af-acbe-bf003b2367ff.png)